esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1831,16 +1866,35 @@ static int bindingCompareFunc(IInterface * const *_itm1, IInterface * const* _it
 
 void getAllEspProcessesSorted(StringArray& processes)
 {
-    Owned<IRemoteConnection> conn = querySDS().connect("/Environment/Software", myProcessSession(), RTM_LOCK_READ , SDS_LOCK_TIMEOUT_DESDL);
-    if (!conn)
-        throw MakeStringException(-1, "Unable to connect to /Environment/Software dali path");
-    Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements("EspProcess");
-    ForEach (*iter)
+    if (isContainerized())
     {
-        IPropertyTree &item = iter->query();
-        processes.append(item.queryProp("@name"));
+        Owned<IPropertyTree> globalConfig = getGlobalConfigSP();
+        if (globalConfig)
```

Comment:
You don't need this if, it will throw an exception if it cant get global config.  But actually you should  just use getGlobalConfigSP() for the getElements below instead of creating a variable.

Reply:
fixed